src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -1682,3 +1682,263 @@ void program::cancel_compilation_context() {
     if (_compilation_context != nullptr)
         _compilation_context->cancel();
 }
+
+void program::save(cldnn::BinaryOutputBuffer& ob) const {
+    std::list<std::pair<primitive_id, primitive_id>> shared_mem_nodes;
+    ob << nodes_map.size();
+    for (auto& node : nodes_map) {
+        ob.setKernelImplParams(node.second->get_kernel_impl_params().get());
+
+        if (node.second->is_type<data>() && node.second->as<data>().get_primitive()->mem == nullptr) {
+            auto& data_node = node.second->as<data>();
+            if (data_node.get_attached_memory_ptr() == nullptr) {
+                ob << false;
+                continue;
+            } else {
+                node.second->as<data>().typed_desc()->mem = data_node.get_attached_memory_ptr();
+            }
+        }
+        ob << true;
+
+        ob << node.second->desc;
+
+        // TopK
+        if (node.second->is_type<mutable_data>()) {
+            std::string md_write = "_md_write";
+            if (node.first.length() > md_write.length() &&
+                node.first.compare(node.first.length() - md_write.length(), md_write.length(), md_write) == 0) {
+                primitive_id argmax_mutable_id_r = node.first.substr(0, node.first.length() - md_write.length()) + ".out1";
```

Comment:
This part is only for TopK operation. The TopK operation is composed of three primitives, and two of them shares the same mutable data as shown below. Because this part is TopK-specific, I used the same variable names as in the TopK operation creator.
https://github.com/openvinotoolkit/openvino/blob/f84d1c531aba01963421163919985f760c36f2dd/src/plugins/intel_gpu/src/plugin/ops/topk.cpp#L68-L99

Reply:
Thank you for your comments. I'll update this part in a more general way. Thank you.

Reply:
I updated this part to handle `mutable_data` nodes in a general way. Thank you.